apps/els_lsp/src/els_compiler_diagnostics.erl

```diff
@@ -202,6 +204,436 @@ diagnostic(Range, Module, Desc, Severity) ->
 format_error(Str) ->
   Str.
 
+
+%-----------------------------------------------------------------------
+
+%% @doc Return a unique code for each of the possible errors returned by the
+%% compiler or its subsystems.
+
+-spec make_code(atom(), any()) -> binary().
+
+%% This file
+make_code(els_compiler_diagnostics, _) ->
```

Comment:
We can proceed like this, but we may consider refactoring things a bit. For example, we could move the codes to a separate module/header and maybe derive the initial letter from the module itself.